src/gui/editors/AutomationEditor.cpp

```diff
@@ -492,162 +419,222 @@ void AutomationEditor::drawLine( int x0In, float y0, int x1In, float y1 )
-			}
+					// If we are pressing shift, make a line of nodes from the last
+					// clicked position to this position
+					if (mouseEvent->modifiers() & Qt::ShiftModifier)
+					{
+						drawLine(m_drawLastTick, m_drawLastLevel,
+							posTicks, level);
 
-			if (mouseEvent->button() == Qt::LeftButton)
-			{
-				m_mouseDownLeft = true;
-			}
-			if( mouseEvent->button() == Qt::RightButton )
-			{
-				m_mouseDownRight = true;
-			}
+						// Update the last clicked position for the next time
+						m_drawLastTick = posTicks;
+						m_drawLastLevel = level;
 
-			// left button??
-			if( mouseEvent->button() == Qt::LeftButton &&
-							m_editMode == DRAW )
-			{
-				m_pattern->addJournalCheckPoint();
-				// Connect the dots
-				if( mouseEvent->modifiers() & Qt::ShiftModifier )
-				{
-					drawLine( m_drawLastTick,
-							m_drawLastLevel,
-							pos_ticks, level );
-					m_mouseDownLeft = false;
-				}
-				m_drawLastTick = pos_ticks;
-				m_drawLastLevel = level;
+						// Changes the action to drawing a line of nodes
+						m_action = DRAW_LINE;
+					}
+					else // No shift, we are just creating/moving nodes
+					{
+						// If the place we clicked had no nodes
+						if (clickedNode == tm.end())
+						{
+							// We create a new node and start dragging it
+							MidiTime valuePos(posTicks);
 
-				// did it reach end of map because
-				// there's no value??
-				if( it == time_map.end() )
-				{
-					// then set new value
-					MidiTime value_pos( pos_ticks );
-
-					MidiTime new_time =
-						m_pattern->setDragValue( value_pos,
-									level, true,
-							mouseEvent->modifiers() &
-								Qt::ControlModifier );
-
-					// reset it so that it can be used for
-					// ops (move, resize) after this
-					// code-block
-					it = time_map.find( new_time );
+							// Starts actually moving/draging the node
+							MidiTime newTime = m_pattern->setDragValue(valuePos,
+								level, true, mouseEvent->modifiers() & Qt::ControlModifier);
+
+							// Set the iterator to our newly created node so we can use it later
+							// for the m_moveXOffset calculation
+							clickedNode = tm.find(newTime);
+						}
+						else // If we clicked over an existing node
+						{
+							// Simply start moving/draging it
+							MidiTime newTime = m_pattern->setDragValue(MidiTime(POS(clickedNode)),
+								level, true, mouseEvent->modifiers() & Qt::ControlModifier);
+
+							// We need to update our iterator because setDragValue removes the node that
+							// is being dragged, so if we don't update it we have a bogus iterator
+							clickedNode = tm.find(newTime);
+						}
+
+						// Set the action to MOVE_VALUE so moveMouseEvent() knows we are moving a node
+						m_action = MOVE_VALUE;
+
+						// Calculate the offset from the place the mouse click happened in comparison
+						// to the center of the node
+						int alignedX = (int) ((float) ((POS(clickedNode) - m_currentPosition) * m_ppb) /
+									MidiTime::ticksPerBar());
```

Comment:
With 4 space tabs, this only hits 94 characters wide.
```suggestion
						int alignedX = static_cast<int>(
							static_cast<float>((POS(clickedNode) - m_currentPosition) * m_ppb)
							/ MidiTime::ticksPerBar()
						);
```